tests/GeneralStateTestsRunner.js

```
@@ -100,6 +100,9 @@ function runTestCase (options, testData, t, cb) {
           tx: tx,
           block: block
         }, function (err, r) {
+          if (err) {
+            console.log(err)
```

Same question as above

---

updated

---

I definitely agree with you that its better to have tests fail explicitly. In this case however, tests are correct: they are testing that the VM functions correctly when input is invalid.

E.g. current state root is `0xdeadbeef`, when the VM is given an invalid tx (sender doesn't have enough gas), `runTx` throws an error. The tests later check that the state root shouldn't have changed for an invalid tx.

---

make sense - updated!